tls/s2n_resume.c

```diff
@@ -228,27 +228,53 @@ static int s2n_client_serialize_resumption_state(struct s2n_connection *conn, st
     return 0;
 }
 
-static S2N_RESULT s2n_tls12_client_deserialize_session_state(struct s2n_connection *conn, struct s2n_stuffer *from)
+static S2N_RESULT s2n_tls12_client_deserialize_session_state(struct s2n_connection *conn,
+        struct s2n_blob *ticket, struct s2n_stuffer *from)
 {
     RESULT_ENSURE_REF(conn);
     RESULT_ENSURE_REF(from);
 
```

Comment:
```suggestion
RESULT_ENSURE_REF(ticket);
```

Reply:
No, because ticket can be NULL if we're dealing with a session id. I could update the session id deserialization logic to pass an empty blob instead of a NULL, but that'd involve updating a bunch of tests and also probably mean the server side logic should also work with an empty blob instead of a NULL, which would also require updating a bunch of tests. I'm not sure the complication is worth it :/